Client/Coordinators/Library/DownloadsCoordinator.swift

```diff
@@ -0,0 +1,64 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+protocol DownloadsNavigationHandler: AnyObject {
+    /// Handles the possibile navigations for a file.
+    /// The source view is the view used to display a popover for the share controller.
+    func handleFile(_ file: DownloadedFile, sourceView: UIView)
+}
+
+class DownloadsCoordinator: BaseCoordinator, DownloadsNavigationHandler, UIDocumentInteractionControllerDelegate {
+    // MARK: - Properties
+
+    private weak var parentCoordinator: LibraryCoordinatorDelegate?
+
+    // MARK: - Initializers
+
+    init(
+        router: Router,
+        parentCoordinator: LibraryCoordinatorDelegate?
+    ) {
+        self.parentCoordinator = parentCoordinator
+        super.init(router: router)
+    }
+
+    // MARK: - DownloadsNavigationHandler
+
+    func handleFile(_ file: DownloadedFile, sourceView: UIView) {
+        if file.mimeType == MIMEType.Calendar {
+            let docController = UIDocumentInteractionController(url: file.path)
+            docController.delegate = self
+            docController.presentPreview(animated: true)
+            return
+        }
+
+        guard file.canShowInWebView
+        else {
+            shareFile(file, sourceView: sourceView)
+            return
+        }
+        parentCoordinator?.libraryPanel(didSelectURL: file.path, visitType: .typed)
+    }
+
+    private func shareFile(_ file: DownloadedFile, sourceView: UIView) {
+        let helper = ShareExtensionHelper(url: file.path, tab: nil)
```

Comment:
We can use the new Share extension coordinator for this?

Reply:
Yes sure, i was not sure to use it because i thought it would display also the send link to device, that is not needed for a file share, but it depends on the url passed so i will implement it. Ah should i protect the share coordinator with its feature flag in the DownloadsCoordinator ?

Reply:
For the feature flag, it looks like the changes for the Share coordinator is dependent on the library coordinator to be entirely done. We won't release one without the other. I wonder if we should have one flag for both library and share extension changes (or make the share extension depend on the library flag).

Reply:
Ok. If we just protect the share coordinator with his feature flag and we keep the share function that is already implemented as fallback, what do you think ?

Reply:
Sounds good to me! We can clean this up later on 👍

Reply:
Ok perfect thank you